float64 min_distance
float64 angle
